### PR TITLE
Add display of electronic portfolio links for Alma documents

### PR DESCRIPTION
**Pull request by eliotjordan**
Works toward #2273 

Adds a new fields to hold electronic portfolio objects from the 951 or AVE fields in Alma records. The links and labels in the field are displayed in the "Available Online" section of the show page and replace the data retrieved from Umlaut.

- New field `electronic_portfolio_s' is multi-valued and holds an escaped JSON strings. Similar to `electronic_access_1display`
- The JSON has the following format:
  ```
    {
      "desc": "Available from 1869 volume: 1 issue: 1.",
      "title": "Nature",
      "url": "https://na07.alma.exlibrisgroup.com/view/uresolver/01PRI_INST/openurl?u.ignore_date_coverage=true&amp;portfolio_pid=53443322610006421&amp;Force_direct=true",
      "start": "1869",
      "end": "latest"
    }
  ```

### CODE REVIEW

**Comment by kevinreiss** [comment]
That JSON looks on target to me.

**Review by christinach — CHANGES_REQUESTED** [review]
@eliotjordan , @mzelesky and I have reviewed the PR and the related issues about electronic records. The PR looks ok but it would be good to also include a fixture that has an embargo (where there is a 954 field in the publishing record). 

This issue https://github.com/pulibrary/bibdata/issues/1001 describes in depth the electronic record and how we would calculate the embargo in bibdata.
 
We will calculate the start and end years in bibdata but it would good to also include in this PR the example mms_id: 99122412208706421 from this issue https://github.com/pulibrary/bibdata/issues/1001. Using the above mms_id the year range would say 1880-latest without the embargo accounted for. But with the embargo it would say 1880-2015 (current example from catalog: https://catalog.princeton.edu/catalog/857469 Science with embargo).

**Review by hackartisan — CHANGES_REQUESTED** [review]
Looks great; 2 small things

File: app/models/solr_document.rb [hunk 1]
```diff
@@ -116,6 +123,23 @@ def full_arks
     electronic_access_uris.select { |x| x.include?('ark:') }
   end
 
+  # Retrieves sibling documents linked by values on the other_version_s field
+  # @return [Array<SolrDocument>] array of sibling solr documents
+  def sibling_documents
```
[the remark this answers is not in the document]
Comment (eliotjordan) [reply app/models/solr_document.rb]: Fixed

File: spec/features/availability_spec.rb [hunk 2]
```diff
@@ -153,17 +153,27 @@
     end
 
     describe 'Electronic Holdings', js: true do
-      it 'within the online section it does not display links', unless: in_ci? do
+      it 'within the online section, it displays electronic portfolio links', unless: in_ci? do
         stub_holding_locations
         visit '/catalog/99122306151806421'
-        expect(page).not_to have_selector '.availability--online a'
+        expect(page).to have_text '1869 - 1923: Biodiversity Heritage Library Free'
+        expect(page).to have_text ' Available from 1869 volume: 1 issue: 1.'
```
[the remark this answers is not in the document]
Comment (eliotjordan) [reply spec/features/availability_spec.rb]: removed

**Review by hackartisan — APPROVED** [review]

**Comment by eliotjordan** [comment]
@christinach I added some fixture data and a test to show how embargoed dates are displayed. However, the logic that calculates the embargoed date is in bibdata.

File: spec/features/availability_spec.rb [hunk 3]
```diff
@@ -152,18 +152,33 @@
...
+    describe 'Electronic Holdings' do
+      it 'within the online section, it displays electronic portfolio links' do
         visit '/catalog/99122306151806421'
-        expect(page).not_to have_selector '.availability--online a'
+        expect(page).to have_text '1869 - 1923: Biodiversity Heritage Library Free'
+        expect(page).to have_text 'Available from 1869 volume: 1 issue: 1.'
+
+        # Electronic portfolio link with an embargo
+        expect(page).to have_text '1990 - 2020: ProQuest Central'
+
```
Comment (christinach) [inline spec/features/availability_spec.rb:163]: This is ok based on the published fixture in bibdata 
`<datafield tag="954" ind1="" ind2="">
    <subfield code="a">53443322630006421</subfield>
    <subfield code="b">&gt;=</subfield>
    <subfield code="c">1</subfield>
  </datafield>`

I will approve this PR based on the display. 
We need though to improve the calculation for the emabrgo date. I opened issue https://github.com/pulibrary/bibdata/issues/1147 with more information on the embargo date. 
Thank you !

**Review by christinach — APPROVED** [review]

**Pull request merged**